### ISSUE TITLE
Wandb not sync during model.train()

### ISSUE THREAD

**Issue by TaiDuc1001** [ultralytics/ultralytics#17449]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I logged wandb with key
`wandb.login(key=' ')`
Then specify the project and name in train method to activate wandb syncing. 
`model.train(data='data.yaml', epochs=1, batch=512, device='0,1', imgsz=320, project='project name', name='run name')`
But it does not sync at all. Just do a normal training and save runs locally.
![image](https://github.com/user-attachments/assets/4820ab9c-94c3-4581-a0d4-2215f729bc00)

### Environment

Ultralytics 8.3.28 🚀 Python-3.10.14 torch-2.4.0 CUDA:0 (Tesla T4, 15095MiB)
Setup complete ✅ (4 CPUs, 31.4 GB RAM, 5933.9/8062.4 GB disk)

OS                  Linux-5.15.154+-x86_64-with-glibc2.35
Environment         Kaggle
Python              3.10.14
Install             pip
RAM                 31.36 GB
Disk                5933.9/8062.4 GB
CPU                 Intel Xeon 2.00GHz
CPU count           4
GPU                 Tesla T4, 15095MiB
GPU count           2
CUDA                12.3

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.4.0>=1.8.0
torchvision         ✅ 0.19.0>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 5.9.3
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.12.2>=0.11.0
ultralytics-thop    ✅ 2.0.11>=2.0.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
torch               ✅ 2.4.0!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

`model.train(data='data.yaml', epochs=1, batch=512, device='0,1', imgsz=320, project='project name', name='run name')`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @TaiDuc1001, thank you for reaching out to us at Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will be with you shortly to help resolve your issue.

In the meantime, we encourage you to explore our [Docs](https://docs.ultralytics.com) for useful examples and solutions, particularly in the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) sections. These resources contain a wealth of information that might address your questions or help in troubleshooting your issue.

As you mentioned encountering a 🐛 bug with wandb not syncing during training, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/)? This will greatly assist us in diagnosing the issue effectively.

If your question is about custom training ❓, please provide details such as dataset examples or training logs, and ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For interactive support, you can join our Ultralytics community on [Discord](https://ultralytics.com/discord) 🎧, engage in detailed discussions on [Discourse](https://community.ultralytics.com), or participate in our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Meanwhile, ensure you're using the latest version of the package. Upgrade with the following command:

```bash
pip install -U ultralytics
```

Make sure your environment meets the requirements specified [here](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), using [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

We support YOLO in a variety of environments, with all necessary dependencies included:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Refer to our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: Check our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Instructions available in our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates the status of our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests. If green, it confirms all tests are passing for YOLO across various platforms, confirming the software's reliable operation in verified environments.

Thank you for your patience and contribution to continually improving Ultralytics software!

**Comment by Y-T-G** [comment]
`wanfb` is no longer enabled by default.

https://github.com/ultralytics/ultralytics/pull/17164

You can run `!yolo settings wandb=True` to enable it.

**Comment by TaiDuc1001** [comment]
Thanks my problem is fixed.

**Comment by glenn-jocher** [comment]
@TaiDuc1001 great to hear that your issue is resolved! If you have any more questions, feel free to ask.
